torch_sparse/narrow.py

```diff
@@ -1,11 +1,69 @@
 from typing import Tuple
 
+import torch
 from torch_sparse.storage import SparseStorage
 from torch_sparse.tensor import SparseTensor
 
 
 def narrow(src: SparseTensor, dim: int, start: int,
-           length: int) -> SparseTensor:
+           length: int, bidim: False,
+           n_active_nodes: None) -> SparseTensor:
+    r"""It resizes :obj:`src` SparseTensor along one
+    specified dimension, or two dimensions at the same time
+    if the SparseTensor has 2 dimensions. In the
+    latter case, it returns a squared SparseTensor.
+    In the case of 1-dimensional resizing,
+    :obj:'start' and :obj:'lenght' are supported
+    In the case of bidimensional resizing, resizing
+    always starts from 0, and it returns a square matrix;
+    :obj:`src` must be a bidimensional SparseTensor
+    for bidim option to work.
+    Args:
+        src (SparseTensor): SparseTensor to be manipulated
+        dim (int): dimension along which the resizing is done.
+                    Not used in case bidim is True
+        start (int): first position to include in the ouput.
+                    Not used in case bidim is True (set to 0)
+        length (int): last position to include in the output.
+        bidim (bool):  it applies bidimensional and squared narrowing
+                    of the src, which must be a bidimensional SparseTensor.
+        n_active_nodes (int): Required only if bidim option is True;
+                    for efficient manipulation of the rowptr bidim case.
+    """
+    if bidim:
+        if len(src.storage._sparse_sizes) != 2:
+            raise NotImplementedError
+        
+        start = 0
+        if src.storage._rowptr is None:
+            rowptr, col, value = src.csr()
+
+        # rowptr
+        rowptr = torch.narrow(src.storage._rowptr,
+                              0, start, length + 1).clone()
```

Comment:
This would mean that `rowptr` no longer starts with `zero` in case `start != 0`.

Reply:
`start `is set to 0 in case `bidim `is `True`. 
The bidimensional option is thought specifically for the `trim_to_layer` application in PyG which -due to the BFS ordering of the nodes of the graph of each batch-  it always has the target nodes as first nodes. 
So, In that case, the resizing always starts from zero up to `lenght` included.

Would you prefer to have the `narrow_block` function/method agnostic to the specific case, and support the possibility to resize a bidim matrix from positions other than 0?